ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -92,6 +92,13 @@
   private volatile boolean responseComplete = false;
   // marked as true if force close is required because close() was called.
   private volatile boolean forceClose = false;
+  // the ResponseStatus that will be sent (or has been sent) as a part of the response.
+  private volatile ResponseStatus responseStatus = ResponseStatus.Ok;
+  // the response metadata that was actually sent.
+  private volatile HttpResponse finalResponseMetadata = null;
```

Comment:
More than one thread has the opportunity to set it (only one will actually do it in the end) because the piece of code that sets it may be reached via `write()` and `onResponseComplete()`. It can be read by any number of threads.